src/main/java/com/patchworkmc/event/EventSubscriptionChecker.java

```diff
@@ -0,0 +1,70 @@
+package com.patchworkmc.event;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+// Currently patchwork cannot handle @SubscribeEvent for overloaded methods
+public class EventSubscriptionChecker {
+	private static class Entry {
+		public List<SubscribeEvent> subscribeEvents;
+		public List<String> superClasses;
+
+		Entry(List<SubscribeEvent> subscribeEvents, List<String> superClasses) {
+			this.subscribeEvents = subscribeEvents;
+			this.superClasses = superClasses;
+		}
+	}
+
+	private Map<String, Entry> data = new HashMap<>();
+
+	public EventSubscriptionChecker() {
+	}
+
+	public void onClassScanned(
+			String className,
+			List<SubscribeEvent> subscribeEvents,
+			List<String> superClasses
+	) {
+		data.put(className, new Entry(subscribeEvents, superClasses));
+	}
+
+	public void check() {
+		data.forEach((className, entry) -> {
+			Set<String> descriptions = new HashSet<>();
+			gatherSuperClasses(className).distinct().flatMap(
```

Comment:
If `gatherSuperClasses` is changed to just be `gatherEventSubscribers` then this could be a simple loop rather than needing all of this stream code.